### ISSUE TITLE
Error Message for Composable Solids with Return Type None ("@composite_solid '...' has unmapped output 'result'")

### ISSUE THREAD

**Issue by KyleKing** [dagster-io/dagster#4083]
## Summary

I am converting a project from Prefect to Dagster and a function seemed to make sense to implement as a composable solid, but it doesn't have a return value so Dagster threw an exception. I'm not sure if this should be resolved in the same way as #3198 for pipelines (https://dagster.phacility.com/D5048) where Dagster infers that there is no output

```sh
raise DagsterInvalidDefinitionError(
dagster.core.errors.DagsterInvalidDefinitionError: @composite_solid 'solids' has unmapped output 'result'. Remove it or return a value from the appropriate solid invocation.
.venv\lib\site-packages\dagster\cli\workspace\workspace.py:146: UserWarning: Error loading repository location dagster.core.errors.DagsterInvalidDefinitionError: @composite_solid 'solids' has 
unmapped output 'result'. Remove it or return a value from the appropriate solid invocation.
```

Either way, I'm fine with the explicit workaround. Maybe the error message could be updated to recommend specifying `output_defs=[]`?

## Reproduction

```py
@composite_solid
def solids() -> None:
  example = my_solid()
```

## Workaround

```py
@composite_solid(output_defs=[])
def solids() -> None:
  example = my_solid()
```

**Comment by alangenfeld** [comment]
I think this is actually fixed with some changes around how `output_defs` merge with function signature information that just went out in `0.11.6`

**Comment by KyleKing** [comment]
Awesome! I can confirm that this is fixed in `0.11.6`
